### PR TITLE
neon/st1{,q}_*_x{2,3,4}: initial implementation

### PR DESCRIPTION
**Pull request by dgazzoni**
This PR implements st1{,q}_*_x{2,3,4}. It includes tests, whose values were generated on a Raspberry Pi 4 (Cortex-A72 core).

### CODE REVIEW

File: simde/arm/neon/st1_x2.h [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+#include "types.h"
+
+HEDLEY_DIAGNOSTIC_PUSH
+SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
+SIMDE_BEGIN_DECLS_
+
+SIMDE_FUNCTION_ATTRIBUTES
+void
+simde_vst1_f32_x2(simde_float32 ptr[HEDLEY_ARRAY_PARAM(4)], simde_float32x2x2_t val) {
```
Comment (mr-c) [inline simde/arm/neon/st1_x2.h:39]: icc complained about this: `identifier "simde_float32x2x2_t" is undefined`

https://github.com/simd-everywhere/simde/pull/922/checks?check_run_id=3732776539#step:8:928

Comment (dgazzoni) [reply simde/arm/neon/st1_x2.h:41]: I reviewed the relevant code in [simde/types/arm/neon.h](https://github.com/dgazzoni/simde/blob/st1_x234/simde/arm/neon/types.h). It turns out that the `...x{2,3,4}_t` structs are defined in either one of two places: on the block starting [here](https://github.com/dgazzoni/simde/blob/972c99c433b9bfb7afd58facdc491cf4fb038ca6/simde/arm/neon/types.h#L319) or the block starting [here] (https://github.com/dgazzoni/simde/blob/972c99c433b9bfb7afd58facdc491cf4fb038ca6/simde/arm/neon/types.h#L658). The first one is conditional on `SIMDE_ARM_NEON_A32V7_NATIVE`, which certainly isn't the case on icc. As for the second one, it is conditional on `SIMDE_BUG_INTEL_857088`. I assume the bug in question is present in the tested version of icc, so this is why neither block is used, and so the definition can't be found.

Following the pattern found in other files that also use `...x{2,3,4}_t` structs, I made all functions conditional on `SIMDE_BUG_INTEL_857088`, so they won't get compiled under icc.

**Review by mr-c — APPROVED** [review]

**Comment by mr-c** [comment]
@dgazzoni I think there is an include missing?

https://github.com/simd-everywhere/simde/actions/runs/6348516960/job/17245284887#step:9:1068

**Comment by dgazzoni** [comment]
> @dgazzoni I think there is an include missing?
> 
> https://github.com/simd-everywhere/simde/actions/runs/6348516960/job/17245284887#step:9:1072

Looks like you figured it out later, these functions are AArch64-only.

**Comment by mr-c** [comment]
> > @dgazzoni I think there is an include missing?
> > https://github.com/simd-everywhere/simde/actions/runs/6348516960/job/17245284887#step:9:1072
> 
> Looks like you figured it out later, these functions are AArch64-only.

`vst1_f32_x2`, `vst1_s8_x2`, `vst1_s16_x2`, `vst1_s32_x2`, `vst1_s64_x2` (etc.) don't seem to be aarch64-only as per https://documentation-service.arm.com/static/5f7b5205bcda971b14568107?token=

https://github.com/simd-everywhere/simde/actions/runs/6350853438/job/17251163558?pr=922#step:9:1076

**Comment by mr-c** [comment]
@yyctw Since [your PR](https://github.com/simd-everywhere/simde/pull/1080) overlaps with this earlier one, can you help me finish this one first? Maybe you can suggest fixes from your PR for the few compiler errors we saw here

**Comment by yyctw** [comment]
> @yyctw Since [your PR](https://github.com/simd-everywhere/simde/pull/1080) overlaps with this earlier one, can you help me finish this one first? Maybe you can suggest fixes from your PR for the few compiler errors we saw here

Yes, I can help finish this PR first.

**Comment by mr-c** [comment]
Merged as https://github.com/simd-everywhere/simde/pull/1082 ; thank you @dgazzoni and @yyctw !

**Pull request closed without merging**